src/main/java/service/impl/FruitTransactionParserImpl.java

```diff
@@ -0,0 +1,20 @@
+package service.impl;
+
+import model.FruitTransaction;
+import service.FruitTransactionParser;
+
+public class FruitTransactionParserImpl implements FruitTransactionParser {
+    @Override
+    public FruitTransaction parse(String line) {
+        String[] parts = line.split(",");

```

Comment:
Consider validating the length of the `parts` array to ensure that it contains all the expected parts of the transaction before accessing them. This helps to avoid `ArrayIndexOutOfBoundsException` if the input line is not properly formatted.